src/algorithm/nest_cfgs.rs

```diff
@@ -442,7 +443,7 @@ pub(crate) mod test {
         let exit = cfg_builder.exit_block();
         cfg_builder.branch(&tail, 0, &exit)?;
 
-        let h = cfg_builder.finish_hugr()?;
+        let h = cfg_builder.finish_hugr(&ExtensionRegistry::new())?;
```

Comment:
maybe a trait level `finish_core_hugr()` method that takes no inputs and assumes no extensions?

Reply:
or maybe it should actually be `finish_prelide_hugr`

Reply:
`finish_prelude_hugr` is a great idea for this the builder! Also done `finish_prelude_hugr_with_outputs` for DataflowHugr; I'm not sure that that includes BlockBuilder but it handles all the cases occurring in tests.

Reply:
BlockBuilder is its own thing, there is one case in the test. I could add a similar method there but +14/-2 didn't shout out "we need this"....(but maybe I'm being lazy, happy to put it in)